flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/ThreadLocalCache.java

```diff
@@ -75,4 +76,13 @@ protected boolean removeEldestEntry(Map.Entry<K, V> eldest) {
             return this.size() > maxSize;
         }
     }
+
+    public static <K, V> ThreadLocalCache<K, V> of(Function<K, V> creator) {
```

Comment:
did you check the performance of this approach? It adds another virtual method call.

Reply:
Tried this:

```java
@Fork(1)
public class MyBenchmark {

    @State(Scope.Thread)
    public static class MyState {
        public int counter = 0;

        public ThreadLocalCache<Integer, String> threadLocalCacheWithoutLambda = new ThreadLocalCache<Integer, String>() {
            @Override
            public String getNewInstance(Integer integer) {
                Blackhole.consumeCPU(2);
                return integer.toString();
            }
        };

        public ThreadLocalCache<Integer, String> threadLocalCacheWithLambda = ThreadLocalCache.of(i -> {
            Blackhole.consumeCPU(2);
            return i.toString();
        });
    }

    @Benchmark
    public void testWithLambda(MyState myState, Blackhole blackhole) {
        blackhole.consume(myState.threadLocalCacheWithLambda.getNewInstance(myState.counter++));
    }

    @Benchmark
    public void testWithoutLambda(MyState myState, Blackhole blackhole) {
        blackhole.consume(myState.threadLocalCacheWithoutLambda.getNewInstance(myState.counter++));
    }

}
```

And here are the results:

```
Benchmark                       Mode  Cnt         Score         Error  Units
MyBenchmark.testWithLambda     thrpt    5  33187007.195 ± 3843963.071  ops/s
MyBenchmark.testWithoutLambda  thrpt    5  34946890.368 ± 1586855.331  ops/s
```

Can't really say anything about it, as results are too volatile on my machine, and probably if there is difference, it's very little

Reply:
Run also a benchmark myself:
```
Benchmark                       Mode  Cnt      Score     Error   Units
MyBenchmark.testWithLambda     thrpt   30  32893.602 ±  70.254  ops/ms
MyBenchmark.testWithoutLambda  thrpt   30  32856.333 ± 155.288  ops/ms
```

Reply:
And a more exhaustive test with 50 warmup and 50 iterations and 10 forks:
```
Benchmark                       Mode  Cnt      Score     Error   Units
MyBenchmark.testWithLambda     thrpt  500  30191.141 ± 252.373  ops/ms
MyBenchmark.testWithoutLambda  thrpt  500  30954.785 ±  79.428  ops/ms
```

Reply:
ok, then let's skip this micro optimization.